include/sparrow/buffer/buffer_adaptor.hpp

```diff
@@ -390,8 +390,7 @@ namespace sparrow
 
     template <typename To, BufferReference<To> FromBufferRef>
         requires T_is_const_if_FromBufferRef_is_const<FromBufferRef, To>
-    constexpr buffer_adaptor<To, FromBufferRef>::size_type
-    buffer_adaptor<To, FromBufferRef>::size() const noexcept
+    constexpr buffer_adaptor<To, FromBufferRef>::size_type buffer_adaptor<To, FromBufferRef>::size() const
```

Comment:
You can use a conditional noexcept here:

```cpp
 constexpr buffer_adaptor<To, FromBufferRef>::size_type
buffer_adaptor<To, FromBufferRef>::size() const noexcept(!SPARROW_CONTRACTS_THROW_ON_FAILURE)
```

It might be better to always define SPARROW_CONTRACTS_THROW_ON_FAILURE (that is, set it to 0 if the user does not pass it as a cmake arg), or ro define a shorter alias to it.